src/components/cax-companyData.tsx

```diff
@@ -76,7 +76,7 @@ export const CompanyDataCax = ({
   const applicationId = obj['applicationId']
 
   useEffect(() => {
-    dispatch(getCompanyDetailsWithAddress(applicationId))
+    dispatch(getCompanyDetailsWithAddress({applicationId, dispatch}))
```

Comment:
this is the old way to do api calls - also it is considered bad practice to call dispatch from a Redux action. See answers in https://stackoverflow.com/a/41260990/830067

convert the redux code to the new apiSlice way as has been done in the portal frontend and then use

`const { data, error } = useGetCompanyDetailsWithAddressQuery(applicationId)`

outside of the useEffect. We can also handle the error here instead of the unnecessary statusCodeError.ts